src/nvim/message.c

```diff
@@ -3290,39 +3275,31 @@ int vim_vsnprintf(char *str, size_t str_m, char *fmt, va_list ap, typval_T *tvs)
 
         case 's':
         case 'S':
-          str_arg = tvs != NULL ? tv_str(tvs, &arg_idx) : va_arg(ap, char *);
-          if (str_arg == NULL) {
+          str_arg = tvs ? tv_str(tvs, &arg_idx) : va_arg(ap, char *);
+          if (!str_arg) {
             str_arg = "[NULL]";
             str_arg_l = 6;
           }
-          /* make sure not to address string beyond the specified
-           * precision !!! */
+          // make sure not to address string beyond the specified precision
           else if (!precision_specified)
             str_arg_l = strlen(str_arg);
-          /* truncate string if necessary as requested by precision */
+          // truncate string if necessary as requested by precision
           else if (precision == 0)
             str_arg_l = 0;
           else {
-            /* Don't put the #if inside memchr(), it can be a
-             * macro. */
-            /* memchr on HP does not like n > 2^31  !!! */
-            char *q = memchr(str_arg, '\0',
-                precision <= (size_t)0x7fffffffL ? precision
-                : (size_t)0x7fffffffL);
-            str_arg_l = (q == NULL) ? precision
-                        : (size_t)(q - str_arg);
+            // memchr on HP does not like n > 2^31
+            // TODO(elmart): check if this still holds / is relevant
+            char *q = memchr(str_arg, '\0', MIN(precision, 0x7fffffff));
+            str_arg_l = q ? (size_t)(q - str_arg) : precision;
```

Comment:
Yeah... I was wondering about that. Must remember to send a patch upstream if it's legit.